File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        
        LinkedList<Integer> rank = new LinkedList<>(); // push도 되나? 되는 것 같긴한데
        int t = 0;
        for (int i = 0; i < score.length; i++) {
            rank.add(score[i]);
            
            Collections.sort(rank);
            
            if (rank.size() > k)
                rank.removeFirst();
            
            answer[t++] = rank.peekFirst();
        }
        
        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    var answer = [];
    
    let rank = [];
    for (let i = 0; i < score.length; i++) {
        rank.push(score[i]);
        
        rank.sort((a, b) => a-b);
        if (rank.length > k)
            rank.shift();
        
        answer.push(rank[0]);
    }
    
    return answer;
}